src/store.cc

```diff
@@ -1714,15 +1714,14 @@ StoreEntry::startWriting()
     rep->packHeadersUsingSlowPacker(*this);
     mem_obj->markEndOfReplyHeaders();
 
+    // Same-worker collapsing risks end with the receipt of the headers.
+    // SMP collapsing risks remain until the headers are actually cached, but
+    // that event is announced via CF-agnostic disk I/O broadcasts.
+    if (hittingRequiresCollapsing())
+        setCollapsingRequirement(false);
```

Comment:
We can skip hittingRequiresCollapsing() check here - setCollapsingRequirement() already cares about this.

Reply:
Good point. Done in ae49f1c5d1d3686cf9f2e25032f5561415fa62ba.